vector/src/androidTest/java/im/vector/app/ui/robot/ElementRobot.kt

```diff
@@ -120,6 +120,26 @@ class ElementRobot {
                                 .perform(ViewActions.closeSoftKeyboard(), click())
                     }
                 }
+                // at this point we are in a race with the app restarting. The steps that happen are:
+                // - (initially) app has started, app has initial synched
+                // - (restart) app has strted, app has not initial synched
+                // - (racey) app shows some UI but overlays with initial sync ui
+                // - (initial sync finishes) app has started, has initial synched
+
+                // We need to wait for the initial sync to complete; but we can't
+                // use waitForHome() like login does.
+
+		// waitForHome() -- does not work because we have already fufilled the initialSync
+                // so we can racily have an IllegalStateException that we have transitioned from busy -> idle
+                // but never having sent the signal.
+
+                // So we need to not start waiting for an initial sync until we have restarted
+                // then we do need to wait for the sync to complete.
+
+                // Which is convoluted especially as it involves the app state refreshing
+                // so; in order to make this be more stable
+                // I hereby cheat and write:
+                Thread.sleep(30_000)
```

Comment:
to be clear: i feel bad about this but i experimented with various options for ~1day and none were more likely to work than this - handling the restart requirement just makes it hard to test with.

Reply:
this seems fair, it sounds like we're missing registering the sync as an `IdlingResource` in order for espresso to be able to wait for the sync to finish 

there's a helper for `withRetry` if we wanted to poll for the home appearing instead of a blanket 30 second sleep (assuming the original check wasn't providing false positives)

```kotlin
fun waitForHome() {
  // each attempt has a 500ms delay
  withRetry(times = 60)
  ... original logic
  }
}
```

Reply:
The problem is that waitForHome() is using an idling resource on initial sync; but it's idle at two points: Immediately (+/-  thread racing) after we return from getting espresso to click the toggle, and then it's busy for 10-15s until after the app has restarted and done the initial sync.